youtube_dl/extractor/srgssr.py

```diff
@@ -36,133 +59,296 @@ def _get_tokenized_src(self, url, video_id, format_id):
+                formats.append({
+                    'format_id': format_id,
+                    'ext': encoding.lower() if encoding else None,
+                    'url': format_url,
+                    'preference': preference(quality)
+                })
+
+        podcast_keys = ('podcastSdUrl', 'podcastHdUrl')
+        podcast_qualities = ('SD', 'HD')
+        if chapter_data['position'] == 0:
+            for key, quality in zip(podcast_keys, podcast_qualities):
+                if chapter_data.get(key):
                     formats.append({
-                        'format_id': format_id,
-                        'url': asset_url,
+                        'format_id': 'PODCAST-HTTP-%s' % quality,
+                        'url': chapter_data[key],
                         'preference': preference(quality),
-                        'ext': 'flv' if protocol == 'RTMP' else None,
                     })
         self._sort_formats(formats)
 
         return {
             'id': media_id,
             'title': title,
             'description': description,
+            'duration': duration,
             'timestamp': timestamp,
-            'thumbnails': thumbnails,
+            'thumbnail': thumbnail,
+            'subtitles': subtitles,
             'formats': formats,
         }
 
 
 class SRGSSRPlayIE(InfoExtractor):
     IE_DESC = 'srf.ch, rts.ch, rsi.ch, rtr.ch and swissinfo.ch play sites'
-    _VALID_URL = r'https?://(?:(?:www|play)\.)?(?P<bu>srf|rts|rsi|rtr|swissinfo)\.ch/play/(?:tv|radio)/[^/]+/(?P<type>video|audio)/[^?]+\?id=(?P<id>[0-9a-f\-]{36}|\d+)'
-
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            (?:
+                                www|
+                                play
+                            )\.
+                        )?
+                        (?P<bu>
+                            srf|
+                            rts|
+                            rsi|
+                            rtr|
+                            swissinfo
+                        )\.ch/play/
+                        (?:
+                            tv|
+                            radio
+                        )/[^/]+/
+                        (?P<type>
+                            video|
+                            audio
+                        )/[^?]+\?id=
+                        (?P<id>
+                            [0-9a-f\-]{36}|
+                            \d+
+                        )
+                    '''
     _TESTS = [{
+        # ID in url not the same as media ID, no Save button, no description
         'url': 'http://www.srf.ch/play/tv/10vor10/video/snowden-beantragt-asyl-in-russland?id=28e1a57d-5b76-4399-8ab3-9097f071e6c5',
         'md5': 'da6b5b3ac9fa4761a942331cef20fcb3',
         'info_dict': {
-            'id': '28e1a57d-5b76-4399-8ab3-9097f071e6c5',
+            'id': '0d181f8f-07ae-46b2-8d3b-7619c0efb0e3',
```

Comment:
Yes, but youtube-dl extracts the whole episode of these videos (even before my commit), and not the single segments, because there is no explicit download source for the single segments...